lib/request_overrider.js

```diff
@@ -63,35 +63,41 @@ function setRequestHeaders(req, options, interceptor) {
   }
 }
 
+function emitError(req, error) {
+  process.nextTick(() => {
+    req.emit('error', error)
+  })
```

Comment:
This seems slightly over-DRYed and I'm thinking it would be better to duplicate this small bit of code into the `emitError()` method and the `emitErrorOnReq()` function. Would also allow renaming `emitErrorOnReq()` to `emitError()`.